crates/networking/p2p/rlpx/connection.rs

```diff
@@ -75,29 +74,34 @@ impl<S: AsyncWrite + AsyncRead + std::marker::Unpin> RLPxConnection<S> {
         )
     }
 
-    pub async fn initiator(signer: SigningKey, msg: &[u8], stream: S, storage: Store) -> Self {
+    pub async fn initiator(
+        signer: SigningKey,
+        msg: &[u8],
+        stream: S,
+        storage: Store,
+    ) -> Result<Self, RLPxError> {
         let mut rng = rand::thread_rng();
         let digest = Keccak256::digest(&msg[65..]);
-        let signature = &Signature::from_bytes(msg[..64].into()).unwrap();
-        let rid = RecoveryId::from_byte(msg[64]).unwrap();
-        let peer_pk = VerifyingKey::recover_from_prehash(&digest, signature, rid).unwrap();
+        let signature = &Signature::from_bytes(msg[..64].into())?;
```

Comment:
To make this even more resilient, instead of accessing with slicing, let's use the [get method](https://doc.rust-lang.org/std/primitive.slice.html#method.get) in combination with [ok_or](https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or) to turn it into an RLPxError